components/director/internal/domain/api/repository_test.go

```diff
@@ -0,0 +1,327 @@
+package api_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/DATA-DOG/go-sqlmock"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/api/automock"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/kyma-incubator/compass/components/director/internal/persistence"
+	"github.com/kyma-incubator/compass/components/director/internal/repo/testdb"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPgRepository_GetByID(t *testing.T) {
+	// given
+	tenantID := fixTenantID()
+	apiDefID := fixApiDefID()
+	apiDefEntity := *fixFullEntityAPIDefinition(fixApiDefID(), "placeholder")
+
+	sqlxDB, sqlMock := testdb.MockDatabase(t)
+	selectQuery := `^SELECT (.+) FROM "public"."api_definitions" WHERE tenant_id = \$1 AND id = \$2$`
```

Comment:
You don't need to escape that, use `QuoteMeta`

Reply:
I can use it here, because I am using `$` regex special character and `(.+)`.

Reply:
Use `regexp.QuoteMeta`